scipy/stats/_resampling.py

```diff
@@ -349,19 +353,60 @@ def bootstrap(data, statistic, *, n_resamples=9999, batch=None,
     >>> print(std_sample)
     3.9460644295563863
 
-    We can calculate a 90% confidence interval of the statistic using
-    `bootstrap`.
+    The bootstrap is used to approximate the variability we would expect if we
+    were to repeatedly sample from the unknown distribution and calculate the
+    statistic of the sample each time. It does this by repeatedly resampling
+    values *from the original sample* with replacement and calculating the
+    statistic of each resample. This results in a "bootstrap distribution" of
+    the statistic.
 
+    >>> import matplotlib.pyplot as plt
     >>> from scipy.stats import bootstrap
     >>> data = (data,)  # samples must be in a sequence
     >>> res = bootstrap(data, np.std, confidence_level=0.9,
     ...                 random_state=rng)
+    >>> plt.hist(res.bootstrap_distribution, bins=25)
+    >>> plt.title('Bootstrap Distribution')
+    >>> plt.xlabel('statistic value')
+    >>> plt.ylabel('frequency')
+    >>> plt.show()
```

Comment:
For all matplotlib code we want to use the OOP interface so `fig, ax = plt.subplots()`

Reply:
Hmmm I haven't been converted, personally. Did we resolve to do that in all new examples? (I have a really hard time calling "`subplots`" when I just want one plot.)

Reply:
Yes because it's the recommended way from matplotlib. So we should only show what they recommend.